Backend/Remora.Discord.API/API/CDN.cs

```diff
@@ -493,19 +494,20 @@ public static Result<Uri> GetDefaultUserAvatarUrl
         Optional<ushort> imageSize = default
     )
     {
-        return GetDefaultUserAvatarUrl(user.Discriminator, imageFormat, imageSize);
+        return GetDefaultUserAvatarUrl(user.ID, imageFormat, imageSize);
     }
 
     /// <summary>
     /// Gets the CDN URI of the given user's default avatar.
     /// </summary>
-    /// <param name="userDiscriminator">The user's discriminator.</param>
+    /// <param name="userDiscriminatorOrID">The user's discriminator, or their ID.</param>
     /// <param name="imageFormat">The requested image format.</param>
     /// <param name="imageSize">The requested image size. May be any power of two between 16 and 4096.</param>
     /// <returns>A result which may or may not have succeeded.</returns>
+    /// <remarks>The ID of the user should be preferred, as any migrated accounts will have a discriminator of 0, and the result will be wrong.</remarks>
     public static Result<Uri> GetDefaultUserAvatarUrl
     (
-        ushort userDiscriminator,
+        OneOf<Snowflake, ushort> userDiscriminatorOrID,
```

Comment:
Should we still support discriminators here? Won't using the user's ID work in all cases?

Reply:
Discussed on Discord; bots still retain discrims for the foreseeable future